generate-doc.py

```diff
@@ -0,0 +1,168 @@
+    YELLOW = '\033[93m'
+    BLUE = '\033[94m'
+    RESET = '\033[0m'
+
+
+def print_colored(text, color):
+    print(f"{color}{text}{Colors.RESET}")
+
+
+class DocGenerationError(Exception):
+    def __init__(self):
+        self.message = "Error: The documentation is not up to date. Re running pre-commit may help."
+        super().__init__(f"{Colors.RED}{self.message}{Colors.RESET}")
+
+
+def safe_make_dir(directory_path):
+    if not os.path.isdir(directory_path):
+        try:
+            os.makedirs(directory_path)
+        except FileExistsError:
+            shutil.rmtree(directory_path)
+            safe_make_dir(directory_path)
+        except Exception as e:
+            print_colored(
+                f"Error while removing directory '{directory_path}': {e}", Colors.BLUE)
+
+
+def safe_remove_directory(directory_path):
+    if os.path.isdir(directory_path):
+        try:
+            shutil.rmtree(directory_path)
+        except Exception as e:
+            print(f"Error while removing directory '{directory_path}': {e}")
+    else:
+        print(f"Directory '{directory_path}' does not exist.")
+
+
+def copy_file(source_path, destination_path):
+    try:
+        shutil.copy(source_path, destination_path)
+    except FileNotFoundError:
+        print_colored(f"Source file not found.", Colors.RED)
+    except PermissionError:
+        print_colored(
+            "Permission error. Make sure you have the necessary permissions.", Colors.RED)
+    except Exception as e:
+        print_colored(f"An error occurred: {e}", Colors.RED)
+
+
+def remove_formatting(content):
+    # Remove whitespaces and newlines and hyphens
+    return content.replace(" ", "").replace("\n", "").replace("-", "")
+
+
+def files_equal(file1_path, file2_path):
+    with open(file1_path, 'r') as file1, open(file2_path, 'r') as file2:
+        content1 = remove_formatting(file1.read())
+        content2 = remove_formatting(file2.read())
+
+        return content1 == content2
+
+
+def run():
+ # go through actions
+    safe_make_dir("tmps")
+    tmp_action = "tmps/actions"
+    safe_make_dir(tmp_action)
+    action_dir = "actions"
+    changes = []
+    for action_name in os.listdir(action_dir):
+        action_subdir_path = os.path.join(action_dir, action_name)
+        output_dir_action = f"docs/references/actions/{action_name}"
+
+        # Itterate over subidrs of the actions dir
+        if not os.path.isfile(action_subdir_path):
+            action_file = f"{action_subdir_path}/action.y*"
+            action_file1 = f"{action_subdir_path}/action.yml"
+            action_file2 = f"{action_subdir_path}/action.yaml"
+
+            # exclude action directory without an action.yaml file
+            if os.path.exists(action_file1) or os.path.exists(action_file2):
+                # create docu in tmp dir
+                tmp_docu_output_dir = os.path.join(
+                    tmp_action, action_name)
+                safe_make_dir(tmp_docu_output_dir)
```

Comment:
Do we need to use a custom function here that deletes the directory before recreating it or could we also ignore it if it already exists? This would already be supported by `os.makedirs(..., exist_ok=True)` (see https://docs.python.org/3/library/os.html#os.makedirs)